debezium-core/src/main/java/io/debezium/transforms/TimezoneConverter.java

```diff
@@ -0,0 +1,388 @@
+                ZonedDateTime zdtWithZone = zdt.withZoneSameInstant(zoneId);
+                updatedFieldValue = ZonedTimestamp.toIsoString(zdtWithZone, null, 0);
+                break;
+            case ZonedTime.SCHEMA_NAME:
+                OffsetTime offsetTime = OffsetTime.parse((String) fieldValue, DateTimeFormatter.ISO_OFFSET_TIME);
+                OffsetTime offsetTimeWithZone = offsetTime.withOffsetSameInstant(zoneId.getRules().getOffset(Instant.now()));
+                updatedFieldValue = ZonedTime.toIsoString(offsetTimeWithZone, null);
+                break;
+            case MicroTimestamp.SCHEMA_NAME:
+                Instant microInstant = Instant.ofEpochMilli((long) fieldValue / 1_000);
+                LocalDateTime microLocalDateTime = microInstant.atOffset(ZoneOffset.UTC).toLocalDateTime();
+                updatedFieldValue = microLocalDateTime.atZone(zoneId).toInstant().toEpochMilli() * 1_000;
+                break;
+            case NanoTimestamp.SCHEMA_NAME:
+                Instant nanoInstant = Instant.ofEpochMilli((long) fieldValue / 1_000_000);
+                LocalDateTime nanoLocalDateTime = nanoInstant.atOffset(ZoneOffset.UTC).toLocalDateTime();
+                updatedFieldValue = nanoLocalDateTime.atZone(zoneId).toInstant().toEpochMilli() * 1_000_000;
+                break;
+            case Timestamp.SCHEMA_NAME:
+                Instant instant = Instant.ofEpochMilli((long) fieldValue);
+                LocalDateTime localDateTime = instant.atOffset(ZoneOffset.UTC).toLocalDateTime();
+                updatedFieldValue = localDateTime.atZone(zoneId).toInstant().toEpochMilli();
+                break;
+            case org.apache.kafka.connect.data.Timestamp.LOGICAL_NAME:
+                // to-do
+                break;
+        }
+        return updatedFieldValue;
+    }
+
+    private boolean isTimeField(String schemaName) {
+        return schemaName != null && SUPPORTED_TIMESTAMP_LOGICAL_NAMES.contains(schemaName);
+    }
+
+    private void handleStructs(Struct value, Type listType, String matchName, String fieldName) {
+        if (listType == null || matchName == null) {
+            return;
+        }
+
+        Struct before = getStruct(value, Envelope.FieldName.BEFORE);
+        Struct after = getStruct(value, Envelope.FieldName.AFTER);
+
+        if (before != null) {
+            handleValueForFields(Envelope.FieldName.BEFORE, before, listType, matchName, fieldName);
+        }
+        if (after != null) {
+            handleValueForFields(Envelope.FieldName.AFTER, after, listType, matchName, fieldName);
+        }
+    }
+
+    private void handleValueForFields(String structType, Struct structValue, Type listType, String matchName, String fieldName) {
+        Schema schema = structValue.schema();
+        for (org.apache.kafka.connect.data.Field field : schema.fields()) {
+            if ((listType == Type.INCLUDE && field.name().equals(fieldName))
```

Comment:
Please reorder the condition to have `listType == Type.ALL` first as fast resolution.